### PR TITLE
feat: add RangePicker component

### PR DESCRIPTION
**Pull request by pirhoo**
## PR description

This PR implements a new component called `RangePicker` which can be used to wrap an arbitrary HTML element with a "range picker".

## Preview

[Screencast from 12-10-2023 17:35:20.webm](https://github.com/ICIJ/murmur/assets/471176/eab94174-1399-4652-935e-e697fd968ddf)

[Screencast from 12-10-2023 17:35:48.webm](https://github.com/ICIJ/murmur/assets/471176/8bfb041a-7dcc-4922-a925-0cb40fb2aa24)

### CODE REVIEW

File: lib/components/RangePicker.vue [hunk 1]
```diff
@@ -0,0 +1,326 @@
...
+
+<template>
+  <div class="range-picker" :class="classList">
+    <slot />
+    <div class="range-picker__bounds" :style="boundsStyle">
+      <div v-draggable.relative class="range-picker__bounds__overlay" :style="overlayStyle" @dragged="dragBounds"></div>
+      <button v-draggable :style="startBoundStyle" class="range-picker__bounds__start btn" @dragged="dragStartBound">
+        <fa icon="fa-grip-lines-vertical" fixed-width />
+      </button>
+      <button v-draggable class="range-picker__bounds__end btn" :style="endBoundStyle" @dragged="dragEndBound">
```
Comment (caro3801) [inline lib/components/RangePicker.vue:237]: On the button (start and end), I suggest to change the cursor to cursor: 'col-resize' (and keep the cursor as it while dragging.

File: lib/components/RangePicker.vue [hunk 2]
```diff
@@ -0,0 +1,326 @@
...
+      }
+    }
+  },
+  model: {
+    prop: 'value',
+    event: 'update'
+  },
+  props: {
+    /**
+     * Represents the initial values of the range bounds. Should contain two numbers
```
Comment (caro3801) [inline lib/components/RangePicker.vue:66]: In other component props doc we have more direct descriptions without the verbs.
```suggestion
     * Initial values of the range bounds. Should contain two numbers.
```

File: lib/components/RangePicker.vue [hunk 3]
```diff
@@ -0,0 +1,326 @@
...
+     * indicating the start and end of the range.
+     */
+    value: {
+      type: Array as PropType<number[]>,
+      required: true
+    },
+    /**
+     * Defines an offset from the left side of the component
```
Comment (caro3801) [inline lib/components/RangePicker.vue:74]:
```suggestion
     * Offset from the left side of the component
```

File: lib/components/RangePicker.vue [hunk 4]
```diff
@@ -0,0 +1,326 @@
...
+     * where the dragging for the start value begins.
+     */
+    startOffset: {
+      type: [Number as PropType<number>, String as PropType<string>],
+      default: 0
+    },
+    /**
+     * Defines an offset from the right side of the component where
```
Comment (caro3801) [inline lib/components/RangePicker.vue:82]:
```suggestion
     * Offset from the right side of the component where
```

File: lib/components/RangePicker.vue [hunk 5]
```diff
@@ -0,0 +1,326 @@
...
+     * the dragging for the end value ends.
+     */
+    endOffset: {
+      type: [Number as PropType<number>, String as PropType<string>],
+      default: 0
+    },
+    /**
+     * Specifies the number of decimal places to which values should be rounded.
```
Comment (caro3801) [inline lib/components/RangePicker.vue:90]:
```suggestion
     * Number of decimal places to which values should be rounded.
```

File: lib/components/RangePicker.vue [hunk 6]
```diff
@@ -0,0 +1,326 @@
...
+     */
+    precision: {
+      type: Number as PropType<number>,
+      default: 4
+    },
+    /**
+     * Determines the increments in which values should snap. For instance,
```
Comment (caro3801) [inline lib/components/RangePicker.vue:97]:
```suggestion
     * Snap increment value. For instance,
```

File: lib/components/RangePicker.vue [hunk 7]
```diff
@@ -0,0 +1,326 @@
...
+     * if snap is 0.1, values will snap to 0, 0.1, 0.2, and so on.
+     */
+    snap: {
+      type: Number as PropType<number>,
+      default: 0.0001
+    },
+    /**
+     * Specifies the minimum allowed distance between the two range bounds to ensure they
```
Comment (caro3801) [inline lib/components/RangePicker.vue:105]:
```suggestion
     * Minimum distance between the two range bounds to ensure they
```

File: lib/components/RangePicker.vue [hunk 8]
```diff
@@ -0,0 +1,326 @@
...
+    /**
+     * Sets the visual variant/style of the component. Expected to be one
+     * of the predefined Bootstrap theme (e.g., 'primary', 'secondary', etc).
+     */
+    variant: {
+      type: String as PropType<Variant>,
+      default: 'primary'
+    },
+    /**
+     * Determines whether the edges of the range should be rounded. If
```
Comment (caro3801) [inline lib/components/RangePicker.vue:121]:
```suggestion
     * Rounds corner edges of the range boundaries. If
```

File: lib/components/RangePicker.vue [hunk 9]
```diff
@@ -0,0 +1,326 @@
...
+    /**
+     * Specifies the minimum allowed distance between the two range bounds to ensure they
+     * don't get too close to each other.
+     */
+    minDistance: {
+      type: Number as PropType<number>,
+      default: 0.01
+    },
+    /**
+     * Sets the visual variant/style of the component. Expected to be one
```
Comment (caro3801) [inline lib/components/RangePicker.vue:113]: Variant style of the component. Expected to be one

File: lib/components/RangePicker.vue [hunk 10]
```diff
@@ -0,0 +1,326 @@
...
+/**
+ * A component to wrap an HTML element with a range picker overlay.
+ */
+export default defineComponent({
+  name: 'RangePicker',
+  components: {
+    Fa
+  },
+  directives: {
+    draggable: {
```
Comment (caro3801) [inline lib/components/RangePicker.vue:19]: I'm not familiar with directives: are the event listeners applied on the $el (like mousedown) removed on destroy?

**Review by caro3801 — APPROVED** [review]
Looks great, well tested and it's working smoothly. 
UX idea if it's interesting to add it on this component: on hover of the range add a display clue to show the complete range area available.
For cursor suggestion it would be better to go all in for one type (your preference): hand cursors (grab/release..) or with arrows (cross, col resize..)

File: lib/components/RangePicker.vue (hunk 10 above)
Comment (pirhoo) [reply lib/components/RangePicker.vue:19]: Yep, event listeners are garbage collected by the web browser when destroying the element. So no need to destroy them explicitly.

File: lib/components/RangePicker.vue (hunk 1 above)
Comment (pirhoo) [reply lib/components/RangePicker.vue:237]: Changing, thanks for the suggestions!

File: lib/components/RangePicker.vue (hunk 10 above)
Comment (caro3801) [reply lib/components/RangePicker.vue:19]: Neat! thanks

**Pull request merged**